src/api/app/controllers/staging/staged_requests_controller.rb

```diff
@@ -15,22 +15,27 @@ def index
   def create
     authorize @staging_project, :update?
 
-    result = ::Staging::StagedRequests.new(
+    if params[:remove_exclusion]
+      ::Staging::RequestExcluder
+        .new(requests_xml_hash: { number: @request_numbers },
+             staging_workflow: @staging_workflow)
+        .destroy!
+    end
+
+    ::Staging::StagedRequests.new(
       request_numbers: @request_numbers,
       staging_workflow: @staging_workflow,
       staging_project: @staging_project,
       user_login: User.session!.login
-    ).create
+    ).create!
 
-    if result.valid?
-      render_ok
-    else
-      render_error(
-        status: 400,
-        errorcode: 'invalid_request',
-        message: "Assigning requests to #{@staging_project} failed: #{result.errors.to_sentence}."
-      )
-    end
+    render_ok
+  rescue Staging::ExcludedRequestNotFound => e
```

Comment:
this is all api, no? then better not use `StandardError` but ` APIError`, then you don't need to rescue at all. Just

```
class ExcludedRequestNotFound < APIError
   setup 'invalid_request'
end
```

And it will render a 400 - and if you throw it, you give it the message

Reply:
and why you care here but not https://github.com/openSUSE/open-build-service/pull/8695 @coolo ? That's what I was suggesting - specially because the code there happens in `show` and `accept`

Reply:
I've applied the changes @coolo suggested. I need to close this PR now. @victor, @coolo we can review error managing later in another issue.

Reply:
please don’t mention-spam. Thanks

Reply:
Oops sorry, my bad. I wanted to reach @vpereira